# E-Commerce Sales Analysis:
This notebook analyzes an e-commerce sales dataset using Python and Pandas.

In [53]:
import pandas as pd
df=pd.read_csv("Day8_Ecommerce_Sales_Dataset.csv")

# Dataset Overview:
The dataset is loaded into a Pandas DataFrame for analysis. Basic information about the dataset is inspected before performing further operations.

In [54]:
df.head(3)

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
0,1001,2026-02-27,Kabir Ali,Hyderabad,South,Electronics,Wireless Headphones,3,1499,0,4497.0,3.8,Cash on Delivery
1,1002,2026-01-24,Aditya Verma,Jammu,North,Sports,Running Shoes,1,2799,0,2799.0,4.1,Cash on Delivery
2,1003,2026-04-18,Sara Ahmed,Chennai,South,Electronics,Power Bank,2,1199,15,2038.3,4.6,Cash on Delivery


In [34]:
df.shape

(100, 13)

In [55]:
df.columns

Index(['Order_ID', 'Order_Date', 'Customer', 'City', 'Region', 'Category',
       'Product', 'Quantity', 'Unit_Price', 'Discount_Percent', 'Total_Sales',
       'Rating', 'Payment_Method'],
      dtype='str')

In [35]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Order_ID          100 non-null    int64  
 1   Order_Date        100 non-null    str    
 2   Customer          100 non-null    str    
 3   City              100 non-null    str    
 4   Region            100 non-null    str    
 5   Category          100 non-null    str    
 6   Product           100 non-null    str    
 7   Quantity          100 non-null    int64  
 8   Unit_Price        100 non-null    int64  
 9   Discount_Percent  100 non-null    int64  
 10  Total_Sales       100 non-null    float64
 11  Rating            100 non-null    float64
 12  Payment_Method    100 non-null    str    
dtypes: float64(2), int64(4), str(7)
memory usage: 10.3 KB


# Data Selection:
Selecting relevant columns from the dataset for analysis.

In [56]:
df[["Order_ID","City","Category","Product","Total_Sales"]]

,Order_ID,City,Category,Product,Total_Sales
0,1001,Hyderabad,Electronics,Wireless Headphones,4497.00
1,1002,Jammu,Sports,Running Shoes,2799.00
2,1003,Chennai,Electronics,Power Bank,2038.30
3,1004,Lucknow,Electronics,Mechanical Keyboard,9496.20
4,1005,Jammu,Electronics,Smart Watch,3134.05
...,...,...,...,...,...
95,1096,Kochi,Sports,Dumbbell Set,9495.25
96,1097,Mumbai,Clothing,T-Shirt,679.15
97,1098,Bengaluru,Home & Kitchen,Electric Kettle,5996.00
98,1099,Ahmedabad,Books,Machine Learning Basics,1998.00


# Data Filtering:
Filtering the dataset to find specific orders based on selected conditions.

In [57]:
df.loc[df["Total_Sales"]>5000]

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
3,1004,2026-03-13,Kabir Ali,Lucknow,North,Electronics,Mechanical Keyboard,4,2499,5,9496.20,4.1,Debit Card
7,1008,2026-04-27,Aman Kumar,Jammu,North,Clothing,Hoodie,4,1599,5,6076.20,4.3,Credit Card
9,1010,2026-03-11,Aman Kumar,Kochi,South,Clothing,Jacket,4,2499,10,8996.40,4.7,Credit Card
10,1011,2026-03-22,Rahul Das,Hyderabad,South,Home & Kitchen,Coffee Maker,2,3499,0,6998.00,4.2,UPI
12,1013,2026-03-09,Maryam Khan,Lucknow,North,Clothing,Jacket,5,2499,15,10620.75,4.8,Net Banking
13,1014,2026-02-05,Dev Patel,Jaipur,North,Sports,Running Shoes,3,2799,0,8397.00,4.6,UPI
16,1017,2026-03-17,Sana Malik,Hyderabad,South,Sports,Cricket Bat,3,2499,0,7497.00,4.0,Net Banking
18,1019,2026-05-19,Reyansh Jain,Chennai,South,Clothing,Jacket,3,2499,0,7497.00,4.7,UPI
27,1028,2026-04-19,Sana Malik,Chandigarh,North,Electronics,Wireless Headphones,4,1499,5,5696.20,4.6,UPI
28,1029,2026-01-28,Sara Ahmed,Hyderabad,South,Electronics,Wireless Headphones,4,1499,5,5696.20,4.1,Credit Card


# Sorting:
Sorting orders by total sales to identify the highest-value orders.

In [58]:
df.sort_values("Total_Sales",ascending=False)

,Order_ID,Order_Date,Customer,City,Region,Category,Product,Quantity,Unit_Price,Discount_Percent,Total_Sales,Rating,Payment_Method
48,1049,2026-03-02,Ishita Gupta,Mumbai,West,Sports,Running Shoes,5,2799,0,13995.00,4.0,Net Banking
52,1053,2026-01-08,Ananya Singh,Hyderabad,South,Sports,Running Shoes,5,2799,5,13295.25,4.2,Credit Card
84,1085,2026-01-28,Sana Malik,Hyderabad,South,Home & Kitchen,Coffee Maker,4,3499,10,12596.40,4.3,Net Banking
89,1090,2026-05-13,Aditya Verma,Lucknow,North,Sports,Cricket Bat,5,2499,0,12495.00,3.8,Credit Card
12,1013,2026-03-09,Maryam Khan,Lucknow,North,Clothing,Jacket,5,2499,15,10620.75,4.8,Net Banking
...,...,...,...,...,...,...,...,...,...,...,...,...,...
60,1061,2026-05-12,Alina Shah,Bengaluru,South,Books,Python Programming,1,749,5,711.55,5.0,Cash on Delivery
96,1097,2026-02-02,Arjun Nair,Mumbai,West,Clothing,T-Shirt,1,799,15,679.15,3.5,UPI
73,1074,2026-03-28,Alina Shah,Lucknow,North,Books,Atomic Habits,1,599,10,539.10,5.0,Net Banking
5,1006,2026-01-21,Manya Rao,Kochi,South,Books,The Alchemist,1,499,10,449.10,4.2,Cash on Delivery


# Sales Analysis by Category:
Analyzing total sales for each product category.

In [59]:
df.groupby("Category")["Total_Sales"].sum().sort_values(ascending=False)

Category
Sports            136026.45
Home & Kitchen    113329.90
Electronics        92086.00
Clothing           69028.20
Books              36792.40
Name: Total_Sales, dtype: float64

# Sales Analysis by City:
Analyzing total sales generated by each city.

In [42]:
df.groupby("City")["Total_Sales"].sum().sort_values(ascending=False)


City
Hyderabad      59076.45
Chennai        47493.20
Lucknow        46194.65
Pune           41908.55
Mumbai         41457.80
Kochi          37257.50
Chandigarh     34539.65
Delhi          30882.80
Jaipur         25534.00
Kolkata        20169.00
Jammu          16059.75
Ahmedabad      14788.70
Bengaluru      13524.30
Srinagar       11778.60
Bhubaneswar     6598.00
Name: Total_Sales, dtype: float64

# Sales Analysis by Product:
Analyzing total sales generated by each product.

In [60]:
df.groupby("Product")["Total_Sales"].sum().sort_values(ascending=False)


Product
Coffee Maker               51610.25
Jacket                     47106.15
Cricket Bat                41483.40
Running Shoes              41285.25
Wireless Headphones        33877.40
Dumbbell Set               30184.90
Mechanical Keyboard        26489.40
Electric Kettle            26382.40
Mixer Grinder              25341.55
Smart Watch                15670.25
Yoga Mat                   12765.80
Atomic Habits              11590.65
Football                   10307.10
Water Bottle                9995.70
Wireless Mouse              9574.35
T-Shirt                     7790.25
Hoodie                      7675.20
Python Programming          7527.45
The Alchemist               7485.00
Power Bank                  6474.60
Jeans                       6456.60
Machine Learning Basics     5694.30
Data Science Handbook       4495.00
Name: Total_Sales, dtype: float64

# Sales Analysis by Payment Method:
Analyzing total sales generated through each payment method.

In [61]:
df.groupby("Payment_Method")["Total_Sales"].sum().sort_values(ascending=False)


Payment_Method
Credit Card         106540.25
Debit Card           98064.90
Net Banking          96371.25
UPI                  96163.25
Cash on Delivery     50123.30
Name: Total_Sales, dtype: float64

# Overall Sales Statistics:
Calculating key statistics for the entire dataset.

In [62]:
df["Total_Sales"].sum()


np.float64(447262.95000000007)

In [64]:
df["Total_Sales"].mean()


np.float64(4472.629500000001)

In [65]:
df["Total_Sales"].max()


np.float64(13995.0)

In [66]:
df["Total_Sales"].min()


np.float64(449.1)

In [67]:
df["Quantity"].sum()

np.int64(294)

In [68]:
df["Order_ID"].count()


np.int64(100)

# Overall Sales Statistics Summary:

- Total Sales: ₹447,262.95
- Average Sales per Order: ₹4,472.63
- Highest Sale: ₹13,995
- Lowest Sale: ₹449.10
- Total Quantity Sold: 294 units
- Total Number of Orders: 100

# Order Analysis by Category:
Analyzing the number of orders placed in each product category.

In [69]:
df.groupby("Category")["Order_ID"].count().sort_values(ascending=False)

Category
Sports            25
Books             21
Home & Kitchen    21
Electronics       19
Clothing          14
Name: Order_ID, dtype: int64

## Key Observations

- The dataset contains 100 orders with a total quantity sold of 294 units.
- Total sales amounted to ₹447,262.95, with an average sales value of ₹4,472.63 per order.
- The highest-value order was ₹13,995, while the lowest-value order was ₹449.10.
- 35 orders had total sales above ₹5,000, indicating a significant number of high-value orders.
- Sports was the top-performing category based on total sales.
- Hyderabad generated the highest total sales among the cities.
- Coffee Maker was the top-performing product based on total sales.
- Credit Card was the payment method associated with the highest total sales.
- Sports also had the highest number of orders, with 25 orders, followed by Books and Home and Kitchen with 21 orders each.